### ISSUE TITLE
Reduce false positives

### ISSUE THREAD

**Issue by M1dgard** [afollestad/nock-nock#12]
I get a lot of warnings that my sites are down, but they are due to connection time-outs or TLS handshake errors. (I don't use cellular data so I get this when moving out of Wi-Fi coverage while a check is in progress.)

I would propose:
* upon failure, retry immediately;
* if the internet connection has disappeared in the meantime, don't issue a "site is down" warning;
* only when the second attempt fails too, issue a warning.

**Comment by Salamandar** [comment]
Do you (@afollestad) check the Android APIs for internet connection ? That would reduce the false positives a lot.
When a website is (detected as) down, maybe retry more frequently than other websites ? That could be set in the settings of course.

**Comment by PanderMusubi** [comment]
Of connection monitoring is already implement or when it is implemented and the false positives are still too many, offer perhaps the following option which is by default off: Add a globally available and independent test url such as wikipedia.org (which can be altered) to ping in case of an alarm in order to check whether or not the connection is down. If that ping is OK, test the url from which the alarm came a second time and only report that outcome.

**Comment by Salamandar** [comment]
@PanderMusubi no need ! Android handles all that for us : 
```
    public boolean isConnectedToInternet(){
        ConnectivityManager connectivity = (ConnectivityManager) _context.getSystemService(Context.CONNECTIVITY_SERVICE);
          if (connectivity != null)  {
              NetworkInfo[] info = connectivity.getAllNetworkInfo();
              if (info != null) 
                  for (int i = 0; i < info.length; i++) 
                      if (info[i].getState() == NetworkInfo.State.CONNECTED)
                          return true;
 
          }
          return false;
    }
```

**Comment by M1dgard** [comment]
@Salamandar @PanderMusubi: [It's already implemented](https://github.com/afollestad/nock-nock/blob/8ec5280a01c52f424b7b70bcdf22852730002070/app/src/main/java/com/afollestad/nocknock/services/CheckService.java#L87).

**Comment by Salamandar** [comment]
@M1dgard I don't know when it has been implemented, but I don't have false positives anymore for a week now. ^^

**Comment by M1dgard** [comment]
@Salamandar The last commit on this repo was 23 days ago, so it's not something in the code. If you get less false positives, that's because of external factors.

Please, look around for three seconds before mindlessly commenting.

**Comment by Salamandar** [comment]
… And maybe the release took some time before going up in FDroid/PlayStore ? That's what I thought.

Please, have some consideration before being aggressive :)

**Comment by amo13** [comment]
Thanks a lot for this great app @afollestad  !!!
I would also be very thankful for fewer false positives! I also get them when on wifi...

**Comment by afollestad** [comment]
@amo13 @Salamandar @M1dgard @PanderMusubi I've had that Android API usage in this app since it's first release.

**Comment by afollestad** [comment]
Well, not quite the first release actually, but for quite a long time.

<img width="719" alt="screen shot 2018-01-06 at 8 04 19 pm" src="https://user-images.githubusercontent.com/1820165/34645814-e0ca871a-f31c-11e7-9bc1-d97507084e58.png">

**Comment by monkeypants** [comment]
I have been using nock nock in parallel with other, more complex availability monitoring solutions for 5 days. I get a lot (few times a day) of alarms that are not in evidence with the other monitoring systems. The first thing I do when I get an alarm is to either check the site myself or pull down nock nock to check all sites again.

So far, 100% of alarms have seemed like false alarms to me. No evidence of a current problem, no evidence in problem around that time in other systems, and (unless I didnt figure it out yet) no evidence in nock nock supporting the alarm.

So to me, it looks like the system is generating too much noise. It seems too sensitive. I will probably uninstall it soon, the signal:noise ratio (SNR) is preventing it from being a useful app to me.

I would like it better if it had a better SNR. It might need two things: less noise, somehow to "tune the bias" or reduce false alarms (e.g. confirmation fail before alarm?). And stronger signal: tell me more about what went wrong; simple logging of fails and successes, error codes/messages, etc).

I know you want to keep it simple and that's great, but the SNR is impacting my use quite badly, to the point of making the app something i dont want to use.

**Comment by monkeypants** [comment]
This might be a duplicate of #9

**Comment by afollestad** [comment]
Well, it is open source and free. So feel free to make improvements yourself if you wish. @monkeypants
